### PR TITLE
Create Contest

### PR DESCRIPTION
**Pull request by miguelrodrigues22**
Added Create Contest Mutation to Frontend

### CODE REVIEW

**Review by tiagocardoso98 — APPROVED** [review]

File: pages/__tests__/Profile.test.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  await waitFor(() => {
+    expect(textElement).toBeInTheDocument()
+  })
+})
+
+test('Check if specific text appears on the page', () => {
+  const { getByText } = render(<Profile />)
+  const expectedText = 'profile'
+
+  expect(getByText(expectedText)).toBeInTheDocument()
```
Comment (JdFSilva) [inline pages/__tests__/Profile.test.tsx:20]: use `screen.getByText` for consistency and also:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

File: package.json [hunk 2]
```diff
@@ -61,23 +64,30 @@
...
-    "jest": "^29.5.0",
+    "jest": "^29.6.1",
     "jest-environment-jsdom": "^29.5.0",
     "lint-staged": "^13.2.2",
     "postcss": "^8.4.23",
     "pre-commit": "^1.2.2",
     "prettier": "^2.8.8",
     "pretty-quick": "^3.1.3",
-    "tailwindcss": "^3.3.2"
+    "react-test-renderer": "^18.2.0",
```
Comment (JdFSilva) [inline package.json:88]: is this needed?

File: components/__tests__/Contest.test.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    voting_phase_end: '1977-12-22T06:09:35+00:00',
+    cover_picture: {
+      picture_path: 'https://www.jquery-az.com/html/images/banana.jpg',
+    },
+    status: 'closed',
+  }
+
+  const { getByText } = render(<ContestCard contest={contest} />)
+
+  const title = getByText('Becky Schneider')
```
Comment (JdFSilva) [inline components/__tests__/Contest.test.tsx:23]: use `screen.getByText`
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

File: components/Navbar/CreateContestForm.tsx [hunk 4]
```diff
@@ -54,12 +79,63 @@ export default function CreateContestForm({
...
+          const createdContest = await createContest({
+            variables: { contest },
+          })
+        } catch (error) {
+          console.error(error)
+        }
+      }
+    } catch (error) {
+      console.error(error)
+    }
```
Comment (JdFSilva) [inline components/Navbar/CreateContestForm.tsx:138]: why 2 try/catch? the outer one already catches every error

File: components/Navbar/CreateContestForm.tsx [hunk 5]
```diff
@@ -54,12 +79,63 @@ export default function CreateContestForm({
...
+        picture_path: cover_picture,
+      }
+
+      const createdPicture = await createPicture({
+        variables: { picture },
+      })
+
+      contestValues.cover_picture = {
+        id: createdPicture.data.create_picture.id,
+      }
```
Comment (JdFSilva) [inline components/Navbar/CreateContestForm.tsx:111]: is this doing anything? `contestValues` isn't used after this

File: components/Navbar/CreateContestForm.tsx [hunk 6]
```diff
@@ -3,21 +3,45 @@ import { Formik, Field, Form } from 'formik'
...
+  const [createPicture] = useMutation(CREATE_PICTURE)
   const [dateOptions, setdateOptions] = useState('')
   const [uploadDate, setuUploadDate] = useState<Date | null>(null)
   const [votingDate, setVotingDate] = useState<Date | null>(null)
 
+  const [automatedDates, setAutomatedDates] = useState<boolean>(false)
+
+  useEffect(() => {
+    setAutomatedDates(automatedDates)
+    console.log('automatedDates', automatedDates)
```
Comment (JdFSilva) [inline components/Navbar/CreateContestForm.tsx:40]: remove this

File: components/Navbar/CreateContestForm.tsx [hunk 7]
```diff
@@ -54,12 +79,63 @@ export default function CreateContestForm({
...
+            prize,
+            upload_phase_start,
+            upload_phase_end,
+            voting_phase_end,
+            automated_dates,
+            // To be replaced when authentication exists
+            created_by: 'test@test.com',
+          }
+
+          contest.cover_picture = createdPicture.data.create_picture.id
```
Comment (JdFSilva) [inline components/Navbar/CreateContestForm.tsx:127]: why is `cover_picture` separate? can't you just add it to the interface and to the object like every other property of the contest?

File: babel.config.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  presets: ['next/babel', '@babel/preset-flow'],
+  plugins: [
+    'babel-plugin-styled-components',
+    '@babel/plugin-proposal-class-properties',
+  ],
+}
```
[the remark this answers is not in the document]
Comment (miguelrodrigues22) [reply babel.config.js]: we need that for testing, but I just moved all test related code to a different PR

File: components/__tests__/Contest.test.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (miguelrodrigues22) [reply components/__tests__/Contest.test.tsx]: done in a different PR

File: components/Navbar/CreateContestForm.tsx (hunk 4 above)
Comment (miguelrodrigues22) [reply components/Navbar/CreateContestForm.tsx:138]: removed

File: components/Navbar/CreateContestForm.tsx (hunk 7 above)
Comment (miguelrodrigues22) [reply components/Navbar/CreateContestForm.tsx:127]: you're right, done

File: package.json (hunk 2 above)
Comment (miguelrodrigues22) [reply package.json]: removed

File: components/Navbar/CreateContestForm.tsx (hunk 5 above)
Comment (miguelrodrigues22) [reply components/Navbar/CreateContestForm.tsx:111]: removed

**Review by JdFSilva — APPROVED** [review]

**Review by FilipaSdS — APPROVED** [review]

**Pull request merged**